hw12-solid/src/main/java/ru/otus/homework/ATMImplementation.java

```diff
@@ -0,0 +1,56 @@
+package ru.otus.homework;
+
+import java.util.*;
+
+public class ATMImplementation implements ATM {
+
+    private final Map<Banknote.Nominal, BanknoteCell> cells = new HashMap<>();
```

Comment:
```suggestion
    private final Map<Banknote.Nominal, BanknoteCell> cells = new EnumMap<>(Banknote.Nominal.class);
```

Лучше для ключей-перечислений использовать EnumMap